.github/workflows/bcny-cmake.yml

```diff
@@ -6,44 +6,88 @@ on:
+        arch: ${{ matrix.arch }}
     - uses: actions/checkout@v4
       with:
         path: ${{ github.workspace }}/SourceCache/cmake
-    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release
-    - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake
+    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }} -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_SYSTEM_NAME=Windows -D CMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake_arch }} -D CMAKE_CROSSCOMPILING=1
+    - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }}
     - run: |
-        cd ${{ github.workspace }}/BinaryCache/cmake
-        cpack -G WIX
-    - uses: actions/upload-artifact@v3
+        cd ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }}
+        cpack -G WIX -D CPACK_SYSTEM_NAME=win64-${{ matrix.cpack_arch }} -D CPACK_TOPLEVEL_TAG=win64-${{ matrix.cpack_arch }} -V
+    - uses: actions/upload-artifact@v4
       with:
-        name: cmake-windows-amd64
-        path: ${{ github.workspace }}/BinaryCache/cmake/*.msi
+        name: cmake-windows-${{ matrix.arch }}
+        path: ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }}/*.msi
     - id: vars
       run: |
-        $Msi = Get-ChildItem ${{ github.workspace }}/BinaryCache/cmake/*.msi | Select-Object -First 1
+        $Msi = Get-ChildItem ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }}/*.msi | Select-Object -First 1
         $Artifact = $Msi.Name
-        $Location = $Msi.FullName
-        $Release = [System.IO.Path]::GetFileNameWithoutExtension($Artifact)
-        Write-Output "artifact=$($Artifact)" >> $Env:GITHUB_OUTPUT
-        Write-Output "location=$($Location)" >> $Env:GITHUB_OUTPUT
+        $Filename = [System.IO.Path]::GetFileNameWithoutExtension($Artifact)
+        # Drop the arch string
+        $Release = $Filename.Remove($Filename.LastIndexOf('-'))
+        Write-Output "artifact_${{ matrix.arch }}=$($Artifact)" >> $Env:GITHUB_OUTPUT
         Write-Output "release=$($Release)" >> $Env:GITHUB_OUTPUT
+
+  release:
+    runs-on: swift-build-windows-latest-32-cores
+    needs: [build]
+
+    steps:
+    - uses: actions/download-artifact@v4
+      with:
+        name: cmake-windows-amd64
+        path: ${{ github.workspace }}/tmp/amd64
+
+    - uses: actions/download-artifact@v4
+      with:
+        name: cmake-windows-arm64
+        path: ${{ github.workspace }}/tmp/arm64
+
     - uses: actions/create-release@v1
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       id: create_release
       with:
         draft: false
         prerelease: false
-        release_name: ${{ steps.vars.outputs.release }}
-        tag_name: ${{ steps.vars.outputs.release }}
+        release_name: ${{ needs.build.outputs.release }}
+        tag_name: ${{ needs.build.outputs.release }}
```

Comment:
I think that we should keep the release name consistent and make it something like "sha-windows-${{ matrix.arch }}".

Reply:
This tries to have a release named "sha-windows" that includes x64 and arm64. That's why the arch is dropped.

Reply:
Ah, I guess that is confusing to me because of the use of the `needs.build.outputs.release`. But making it "sha-windows" is fine, even making it "sha" would be fine (if we wanted to include macOS builds in the future.